xds/internal/balancer/outlierdetection/balancer.go

```diff
@@ -596,13 +606,35 @@ func (b *outlierDetectionBalancer) Target() string {
 // if the SubConn is not ejected.
 func (b *outlierDetectionBalancer) handleSubConnUpdate(u *scUpdate) {
 	scw := u.scw
-	scw.latestState = u.state
-	if !scw.ejected {
-		if scw.listener != nil {
-			b.childMu.Lock()
-			scw.listener(u.state)
-			b.childMu.Unlock()
-		}
+	scw.mu.Lock()
+	scw.healthListener = nil
+	scw.mu.Unlock()
+
+	if scw.listener == nil {
+		return
+	}
+
+	if !scw.healthListenerEnabled {
+		scw.latestDeleveredState = u.state
+	}
```

Comment:
`scw.latestDeleveredState` is used to send subchannel state state update (either through the connectivity listener or through the health listener) from `handleEjectedUpdate`. So, why are we updating it here *only* when health listener is not enabled?